mscore/palettebox.cpp

```diff
@@ -422,5 +427,145 @@ void PaletteBox::changeEvent(QEvent *event)
       if (event->type() == QEvent::LanguageChange)
             retranslate();
       }
+
+QLineEdit* PaletteBox::getSearchBox()
+      {
+      return searchBox;
+      }
+
+bool PaletteBox::noSelection()
+      {
+      for (Palette* p : palettes()) {
+            if (p->getCurrentIdx() != -1)
+                  return false;
+            }
+      return true;
+      }
+
+//---------------------------------------------------------
+//   mousePressEvent
+//---------------------------------------------------------
+
+void PaletteBox::mousePressEvent(QMouseEvent* ev, Palette* p1)
+      {
+      for (Palette* p : palettes()) {
+            QList<PaletteCell*> cells = p->getDragCells();
+            if (cells.isEmpty())
+                  continue;
+            if (p->getCurrentIdx() != -1) {
+                  p->setCurrentIdx(-1);
+                  p->update();
+                  }
+            }
+      p1->setCurrentIdx(p1->idx(ev->pos()));
+      p1->update();
+      searchBox->setFocus();
+      }
+
+//---------------------------------------------------------
+//   keyPressEvent
+//---------------------------------------------------------
+
+void PaletteBox::keyPressEvent(QKeyEvent *event)
```

Comment:
Be careful, keyPressEvent is a function inherited from QWidget http://doc.qt.io/qt-5/qwidget.html#keyPressEvent. Maybe you want to call this one differently??